### PR TITLE
Updated RELEASES.md for 1.35.0

### PR DESCRIPTION
**Pull request by XAMPPRocky**
[Rendered](https://github.com/XAMPPRocky/rust/blob/master/RELEASES.md)

r? @Mark-Simulacrum 

cc @rust-lang/release @steveklabnik

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,96 @@
+Version 1.35.0 (2019-05-23)
+==========================
+
+Language
+--------
+- [`FnOnce`, `FnMut`, and the `Fn` traits are now implemented for `Box<FnOnce>`,
+  `Box<FnMut>`, and `Box<Fn>` respectively.][59500]
+- [You can now coerce closures into unsafe functions.][59580] e.g.
```
Comment (Centril) [inline RELEASES.md]:
```suggestion
- [You can now coerce closures into unsafe function pointers.][59580] e.g.
```

**Review by Centril — CHANGES_REQUESTED** [review]

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,96 @@
+Version 1.35.0 (2019-05-23)
+==========================
+
+Language
+--------
+- [`FnOnce`, `FnMut`, and the `Fn` traits are now implemented for `Box<FnOnce>`,
```
Comment (Centril) [inline RELEASES.md]: I'm debating whether we should expand this bullet with "why should you care" information... On the one hand, it is probably the most important change this release. On the other, the bullet has all information that is sufficient from a "this is what happened" POV.

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,96 @@
...
+
+
+Libraries
+---------
+- [`Thread` will now show its id in `Debug` output.][59460]
+- [`StdinLock`, `StdoutLock`, and `StderrLock` now implement `AsRawFd`.][59512]
+- [`alloc::System` now implements `Default`.][59451]
+- [Expanded `Debug` output (`{:#?}`) for structs now has a trailing comma on the
+  last field.][59076]
+- [`char::{ToLowercase, ToUppercaseImplement}` now
```
Comment (jonas-schievink) [inline RELEASES.md]:
```suggestion
- [`char::{ToLowercase, ToUppercase}` now
```

**Review by jonas-schievink — CHANGES_REQUESTED** [review]

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,96 @@
+Version 1.35.0 (2019-05-23)
+==========================
+
+Language
+--------
+- [`FnOnce`, `FnMut`, and the `Fn` traits are now implemented for `Box<FnOnce>`,
+  `Box<FnMut>`, and `Box<Fn>` respectively.][59500]
+- [You can now coerce closures into unsafe function pointers.][59580] e.g.
+  ```rust
+  unsafe fn call_unsafe(func: unsafe fn() -> ()) -> () {
```
Comment (taiki-e) [inline RELEASES.md]:
```suggestion
  unsafe fn call_unsafe(func: unsafe fn()) {
```

File: RELEASES.md [hunk 5]
```diff
@@ -1,3 +1,96 @@
...
+Compiler
+--------
+- [Added the `armv6-unknown-freebsd-gnueabihf` and
+  `armv7-unknown-freebsd-gnueabihf` targets.][58080]
+- [Added the `wasm32-unknown-wasi` target.][59464]
+
+
+Libraries
+---------
+- [`Thread` will now show its id in `Debug` output.][59460]
```
Comment (sfackler) [inline RELEASES.md]:
```suggestion
- [`Thread` will now show its ID in `Debug` output.][59460]
```

File: RELEASES.md (hunk 2 above)
Comment (XAMPPRocky) [reply RELEASES.md]: This is something that could be expanded on in the blog post.

**Comment by Mark-Simulacrum** [comment]
This is looking good to me; let's aim to get it landed this week so we're ready for release next week.

**Review by BatmanAoD — APPROVED** [review]

File: RELEASES.md [hunk 6]
```diff
@@ -1,3 +1,101 @@
...
+
+Cargo
+-----
+- [You can now set `cargo:rustc-cdylib-link-arg` at build time to pass custom
+  linker arguments when building a `cdylib`.][cargo/6298] Its usage is highly
+  platform specific.
+  
+Misc
+----
+- [The 1.35.0 rust toolchain is now available natively for musl based distros.][58575]
```
Comment (jonas-schievink) [inline RELEASES.md]:
```suggestion
- [The Rust toolchain is now available natively for musl based distros.][58575]
```

**Review by emilyalbini — CHANGES_REQUESTED** [review]
Can you rebase on master and add the CVE number to the 1.34.2 release?

**Comment by jonas-schievink** [comment]
Closing in favor of https://github.com/rust-lang/rust/pull/60837 (since I can't push to this fork)

**Pull request closed without merging**